wpilibc/src/main/native/cpp/livewindow/LiveWindow.cpp

```diff
@@ -175,6 +175,10 @@ void LiveWindow::SetEnabled(bool enabled) {
   std::lock_guard<wpi::mutex> lock(m_impl->mutex);
   if (m_impl->liveWindowEnabled == enabled) return;
   Scheduler* scheduler = Scheduler::GetInstance();
+  m_impl->startLiveWindow = enabled;
+  m_impl->liveWindowEnabled = enabled;
+  // Force table generation now to make sure everything is defined
+  UpdateValues();
```

Comment:
This will deadlock on the mutex.

Reply:
Is the lock not reentrant?

Reply:
Nope. In C++ they are not. 

Change the `lock_guard` to a `unique_lock`, and unlock then lock around the UpdateValues call

```
lock.unlock();
UpdateValues();
lock.lock();
```

Reply:
Does `UpdateValues` also need to be changed to `unique_lock` as well?

Reply:
No

Reply:
I think the right fix is to use a recursive_mutex instead of unlocking?

Reply:
recursive_mutex only makes sense imo when actual recursion is going on. Otherwise, having a non-locking common function protected by a lock in whatever calls it is the accepted way to make a non-recursive mutex work here.